errors/TypesDoNotUnify.md

```diff
@@ -33,6 +33,9 @@ In the example above, the types `String` and `Int` can never be made equal, henc
 - Look carefully at the error, especially the information at the end. Usually, it will help to narrow down the offending expression. For example, in the error message above, we are told that the error occurred "while checking that expression `"test"` has type `Int`".
 
 ## Notes
+### Arrays
+
+Unlike in Javascript, all elements of a PureScript array must be the same type. For example, ```[true, false]``` or ```[1, 2, 3]``` work fine but ```[1, false]``` does not.
```

Comment:
I would just say that the compiler unifies the inferred types of the array elements to make sure they're the same. Although, there are lots of uses of unification in the compiler, and I don't know if we need to give an example for each one.